nemo/collections/asr/parts/utils/decoder_timestamps_utils.py

```diff
@@ -0,0 +1,761 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import math
+import sys
+from datetime import datetime
+from typing import Dict, List, Tuple, Type, Union
+
+import librosa
+import numpy as np
+import soundfile as sf
+import torch
+from omegaconf import OmegaConf
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.metrics.wer import WER
+from nemo.collections.asr.metrics.wer_bpe import WERBPE
+from nemo.collections.asr.models import EncDecCTCModel, EncDecCTCModelBPE
+from nemo.collections.asr.parts.utils.speaker_utils import audio_rttm_map, get_uniqname_from_filepath
+from nemo.collections.asr.parts.utils.streaming_utils import AudioFeatureIterator, FrameBatchASR
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+from nemo.utils import logging
+
+__all__ = ['ASR_TIMESTAMPS']
+
+try:
+    from pyctcdecode import build_ctcdecoder
+except ImportError:
+    logging.info("pyctcdecode is not installed. You must install kenlm and pyctcdecode to use language models")
```

Comment:
Check for pyctcdecode only when LM scoring being requested by user (Line: 373) if not found trough error otherwise with comments to install from certain location manually.

Reply:
Need to discuss this.